golem-test-framework/src/components/k8s.rs

```diff
@@ -0,0 +1,441 @@
+                Ok(hostname) => return hostname,
+                Err(e) => {
+                    error!("Can't get hostname for {name}: {e:?}");
+                    tokio::time::sleep(Duration::from_secs(1)).await;
+                }
+            }
+        }
+    }
+
+    fn ingress_hostname(ingress: &Ingress, service_name: &str) -> anyhow::Result<String> {
+        let hostname = ingress
+            .status
+            .as_ref()
+            .ok_or(anyhow!("No ingress status for {service_name}"))?
+            .load_balancer
+            .as_ref()
+            .ok_or(anyhow!("No load balancer for {service_name}"))?
+            .ingress
+            .as_ref()
+            .ok_or(anyhow!("No ingress for {service_name}"))?
+            .first()
+            .as_ref()
+            .ok_or(anyhow!("Empty ingress for {service_name}"))?
+            .hostname
+            .as_ref()
+            .ok_or(anyhow!("No ingress hostname for {service_name}"))?
+            .to_string();
+
+        Ok(hostname)
+    }
+
+    #[cfg(not(target_os = "linux"))]
+    async fn run_minikube_service(
+        service_name: &str,
+        namespace: &K8sNamespace,
+    ) -> (Url, Option<Child>) {
+        use std::process::Stdio;
+        use tokio::io::{AsyncBufReadExt, AsyncReadExt, BufReader};
+
+        debug!(
+            "Launching minikube service --namespace={} --url {}",
+            namespace.0, service_name
+        );
+
+        let mut attempts = 0;
+        loop {
+            let mut child = Command::new("minikube")
+                .arg("service")
+                .arg(&format!("--namespace={}", namespace.0))
+                .arg("--url")
+                .arg(service_name)
+                .stdout(Stdio::piped())
+                .stderr(Stdio::piped())
+                .kill_on_drop(true)
+                .spawn()
+                .expect("Failed to start minikube");
+
+            let stdout = child.stdout.take().expect("Failed to get stdout");
+            let mut stderr = child.stderr.take().expect("Failed to get stderr");
+
+            let mut stdout_reader = BufReader::new(stdout).lines();
+
+            while let Some(line) = stdout_reader
+                .next_line()
+                .await
+                .expect("Failed to read stdout")
+            {
+                debug!("minikube service stdout: {}", line);
+                if let Ok(url) = Url::parse(&line) {
+                    return (url, Some(child));
+                }
+            }
+
+            let mut stderr_string = String::new();
+            stderr
+                .read_to_string(&mut stderr_string)
+                .await
+                .expect("Failed to read stderr");
+
+            debug!("minikube service stderr: {}", stderr_string);
+
+            if stderr_string.contains("SVC_UNREACHABLE") && attempts < 5 {
+                attempts += 1;
+                tokio::time::sleep(Duration::from_secs(1)).await;
+                continue;
```

Comment:
This `continue` it the last line in `loop`. It does nothing.

Reply:
Nice catch! Don't know how that happened :)

Reply:
Copied from the next `loop`. The next `loop` has `break` after `continue`